pkg/apis/crds/karpenter.sh_nodepools.yaml

```diff
@@ -80,7 +80,7 @@ spec:
                               This is required if Schedule is set.
                               This regex has an optional 0s at the end since the duration.String() always adds
                               a 0s at the end.
-                            pattern: ^([0-9]+(m|h)+(0s)?)$
+                            pattern: ^((([0-9]+(h|m))|([0-9]+h[0-9]+m))(0s)?)$
```

Comment:
Is there a simpler way to write this pattern? I could imagine that you could do something like `([1-9](h|m|s)+)|(0s)`

Reply:
The format you suggested would allow 1h2m1h yo happen

Reply:
We also don't want to allow seconds, since this breaches the smallest denominator value of the schedule, which is minutes

Reply:
I could understand disallowing a leading 0 on the front of the duration, but I don't think it's a big deal, since they should really be treated the exact same. Disallowing it means that users can't do something like 0h5m or 0h or 0m or 5h0m. These are all weird in that there's better ways to define it, but it's up to us on how we want to allow/disallow this.

Reply:
Yep, makes sense. I just looked over it quick. Current regex makes sense 👍🏼